src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseSpanBasedStringConcatTests.cs

```diff
@@ -0,0 +1,829 @@
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseSpanBasedStringConcatTests
+    {
+        #region Reports Diagnostic
+        public static IEnumerable<object[]> Data_SingleViolationInOneBlock_CS
+        {
+            get
+            {
+                yield return new[]
+                {
+                    @"var _ = {|#0:foo.Substring(1) + bar|};",
+                    @"var _ = string.Concat(foo.AsSpan(1), bar);"
+                };
+                yield return new[]
+                {
+                    @"var _ = {|#0:foo.Substring(1, 2) + bar|};",
+                    @"var _ = string.Concat(foo.AsSpan(1, 2), bar);"
+                };
+                yield return new[]
+                {
+                    @"var _ = {|#0:foo + bar.Substring(1)|};",
+                    @"var _ = string.Concat(foo, bar.AsSpan(1));"
+                };
+                yield return new[]
+                {
+                    @"var _ = {|#0:foo + bar.Substring(1) + baz|};",
+                    @"var _ = string.Concat(foo, bar.AsSpan(1), baz);"
+                };
+                yield return new[]
+                {
+                    @"var _ = {|#0:foo.Substring(1) + bar.Substring(1) + baz.Substring(1)|};",
+                    @"var _ = string.Concat(foo.AsSpan(1), bar.AsSpan(1), baz.AsSpan(1));"
+                };
+                yield return new[]
+                {
+                    @"var _ = {|#0:foo.Substring(1, 2) + bar + baz + baz.Substring(1, 2)|};",
+                    @"var _ = string.Concat(foo.AsSpan(1, 2), bar, baz, baz.AsSpan(1, 2));"
+                };
+                yield return new[]
+                {
+                    @"Consume({|#0:foo + bar.Substring(1, 2)|});",
+                    @"Consume(string.Concat(foo, bar.AsSpan(1, 2)));"
+                };
+                yield return new[]
+                {
+                    @"var _ = Fwd({|#0:foo.Substring(1) + bar|});",
+                    @"var _ = Fwd(string.Concat(foo.AsSpan(1), bar));"
+                };
+                yield return new[]
+                {
+                    @"var _ = Fwd({|#0:foo.Substring(1) + bar.Substring(1)|});",
+                    @"var _ = Fwd(string.Concat(foo.AsSpan(1), bar.AsSpan(1)));"
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Data_SingleViolationInOneBlock_CS))]
+        public Task SingleViolationInOneBlock_ReportedAndFixed_CS(string testStatements, string fixedStatements)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = CSUsings + CSWithBody(testStatements),
+                FixedCode = CSUsings + CSWithBody(fixedStatements),
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics = { VerifyCS.Diagnostic(Rule).WithLocation(0) }
+            };
+            return test.RunAsync();
```

Comment:
Consider making the tests `async`, then `await test.RunAsync()`.

Reply:
`public async Task TestMethod() => await test.RunAsync();` is semantically identical to 
`public Task TestMethod() => return test.RunAsync();`, but potentially slower. If we needed to await multiple async calls, it would make sense to mark the method as `async`, but since we have just that one `async` call, it makes sense to simply return the `Task` produced by `test.RunAsync()`.